coralproject/talk
# [CORL-513] Permitted Domain Fixes

## What does this PR do?

Requires that permitted domains have a scheme.

## How do I test this PR?

1. On a fresh install, try to set `localhost` as a permitted domain, notice the error.
2. On a already installed version of Coral, try to edit the permitted domains under `/admin/configure/advanced`, trying again to add `localhost` as a permitted domain

## Migration

A migration is required after this PR is merged:

```js
db.tenants.find().forEach(tenant => {
  let allowedDomains = tenant.allowedDomains.reduce(
    (domains, domain) =>
      domain.startsWith("http")
        ? // Domain already is prefixed.
          domains.concat([domain])
        : // Domain needs prefixing.
          domains.concat(["http://" + domain, "https://" + domain]),
    []
  );
  db.tenants.updateOne({ id: tenant.id }, { $set: { allowedDomains } });
});
```
